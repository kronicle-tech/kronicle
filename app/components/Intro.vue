<template>
  <b-jumbotron :header="intro.title" class="my-3">
    <Markdown :markdown="intro.markdown" />
  </b-jumbotron>
</template>

<script lang="ts">
import Vue from 'vue'
import { BJumbotron } from 'bootstrap-vue'
import Markdown from '~/components/Markdown.vue'

interface Intro {
  title: string
  markdown: string
}

export default Vue.extend({
  components: {
    'b-jumbotron': BJumbotron,
    Markdown,
  },
  data() {
    return {
      intro: {
        title: this.$config.intro.title || 'Kronicle',
        markdown: this.$config.intro.markdown
          || 'Kronicle contains information on the organisation\'s components, teams and areas.',
      } as Intro
    }
  },
})
</script>
