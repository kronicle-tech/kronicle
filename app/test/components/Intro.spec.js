import { mount } from '@vue/test-utils'
import Intro from '@/components/Intro.vue'

describe('Intro', () => {
  let config
  let wrapper
  const createWrapper = () => {
    wrapper = mount(Intro, {
      mocks: {
        $config: config,
      },
    })
  }

  beforeEach(() => {
    config = {
      intro: {}
    }
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when intro config is not set', () => {
    test('renders default intro', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<h1 class="display-3">Kronicle</h1>')
      )
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<p>Kronicle contains information on the organisation\'s components, teams and areas.</p>')
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when intro config is set to undefined', () => {
    beforeEach(() => {
      config.intro = {
        title: undefined,
        markdown: undefined,
      }
    })

    test('renders default intro', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<h1 class="display-3">Kronicle</h1>')
      )
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<p>Kronicle contains information on the organisation\'s components, teams and areas.</p>')
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when intro.title config is set', () => {
    beforeEach(() => {
      config.intro = {
        title: 'Test Title',
      }
    })

    test('renders the custom intro title with default markdown', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<h1 class="display-3">Test Title</h1>')
      )
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<p>Kronicle contains information on the organisation\'s components, teams and areas.</p>')
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when intro.markdown config is set', () => {
    beforeEach(() => {
      config.intro = {
        markdown: '*Test Message*',
      }
    })

    test('renders the custom intro markdown with default title', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<h1 class="display-3">Kronicle</h1>')
      )
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<p><em>Test Message</em></p>')
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when all intro config is set', () => {
    beforeEach(() => {
      config.intro = {
        title: 'Test Title',
        markdown: '*Test Message*',
      }
    })

    test('renders the whole custom intro', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<h1 class="display-3">Test Title</h1>')
      )
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<p><em>Test Message</em></p>')
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
