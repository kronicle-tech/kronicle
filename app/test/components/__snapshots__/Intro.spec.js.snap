// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro when all intro config is set renders the whole custom intro 1`] = `
"<div class=\\"jumbotron my-3\\">
  <h1 class=\\"display-3\\">Test Title</h1>
  <!---->
  <div>
    <p><em>Test Message</em></p>
  </div>
</div>"
`;

exports[`Intro when intro config is not set renders default intro 1`] = `
"<div class=\\"jumbotron my-3\\">
  <h1 class=\\"display-3\\">Kronicle</h1>
  <!---->
  <div>
    <p>Kronicle contains information on the organisation's components, teams and areas.</p>
  </div>
</div>"
`;

exports[`Intro when intro config is set to undefined renders default intro 1`] = `
"<div class=\\"jumbotron my-3\\">
  <h1 class=\\"display-3\\">Kronicle</h1>
  <!---->
  <div>
    <p>Kronicle contains information on the organisation's components, teams and areas.</p>
  </div>
</div>"
`;

exports[`Intro when intro.markdown config is set renders the custom intro markdown with default title 1`] = `
"<div class=\\"jumbotron my-3\\">
  <h1 class=\\"display-3\\">Kronicle</h1>
  <!---->
  <div>
    <p><em>Test Message</em></p>
  </div>
</div>"
`;

exports[`Intro when intro.title config is set renders the custom intro title with default markdown 1`] = `
"<div class=\\"jumbotron my-3\\">
  <h1 class=\\"display-3\\">Test Title</h1>
  <!---->
  <div>
    <p>Kronicle contains information on the organisation's components, teams and areas.</p>
  </div>
</div>"
`;
